lib/jackalope/queue.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Jackalope.Queue do
+  use GenServer
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  def push(item) do
+    GenServer.call(__MODULE__, {:push, item})
+  end
+
+  def pop() do
+    GenServer.call(__MODULE__, :pop)
+  end
+
+  @impl GenServer
+  def init(opts) do
+    data_dir = Keyword.get(opts, :data_dir)
+    db_name = Keyword.get(opts, :db_name)
+    queue_name = Keyword.get(opts, :queue_name)
+    {:ok, db} = CubDB.start_link(data_dir: data_dir, name: db_name)
+    {:ok, pid} = CubQ.start_link(db: db, queue: queue_name)
```

Comment:
Having a whole GenServer just for this seems overkill? Is there a specific reason not to put these in `Jackalope` children list and then call `Cubq` directly via the private functions for adding/removing worklist items?

Reply:
I couldn't figure out how to have both CubDB and CubQ be supervised in the Jackalope group while still passing CubDB into CubQ. 

but also, we needed to keep the pid of the queue and felt like a GenServer for CubQ made the most sense

Reply:
CubDB functions a lot off named processes iirc. Did it not work by passing `db: db_name`? (its prob 6's now, but I am curious if maybe this is a limitation of Cubq that should prob be reported/addressed)